docs/amber_script.md

```diff
@@ -229,26 +231,56 @@ value for `START_IDX` is 0. The default value for `COUNT` is the item count of
 vertex buffer minus the `START_IDX`.
 
 ```
+# Run the given |pipeline_name| which must be a `compute` pipeline. The
+# pipeline will be run with the given workgroup |x|, |y|, |z| dimensions.
+# Each of the x, y and z values must be a uint32.
 RUN <pipeline_name> <x> <y> <z>
 
+# Run the given |pipeline_name| which must be a `graphics` pipeline. The
+# rectangle at |x|, |y|, |width|x|height| will be rendered.
 RUN <pipeline_name> \
   DRAW_RECT POS <x_in_pixels> <y_in_pixels> \
   SIZE <width_in_pixels> <height_in_pixels>
 
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data must be attached to the pipeline. A start index of 0 will be used
+# and a count of the number of elements in the vertex buffer.
 RUN <pipeline_name> DRAW_ARRAY AS <topology>
-RUN <pipeline_name> DRAW_ARRAY AS <topology> START_IDX <value>
-RUN <pipeline_name> DRAW_ARRAY AS <topology> START_IDX <value> COUNT <value>
-
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data must be attached to the pipeline. A start index of |value| will be used
+# and a count of the number of elements in the vertex buffer.
+RUN <pipeline_name DRAW_ARRAY AS <topology> START_IDX <value>
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data must be attached to the pipeline. A start index of |value| will be used
+# and a count |count_value| will be used.
+RUN <pipeline> DRAW_ARRAY AS <topology> START_IDX <value> COUNT <count_value>
+
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data and  index data must be attached to the pipeline. The vertices will be
+# drawn using the given |topology|. A start index of 0 will be used and the
+# count will be determined by the size of the index data buffer.
 RUN <pipeline_name> DRAW_ARRAY INDEXED AS <topology>
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data and  index data must be attached to the pipeline. The vertices will be
+# drawn using the given |topology|. A start index of |value| will be used and
+# the count will be determined by the size of the index data buffer.
 RUN <pipeline_name> DRAW_ARRAY INDEXED AS <topology> START_IDX <value>
+# Run the |pipeline_name| which must be a `graphics` pipeline. The vertex
+# data and  index data must be attached to the pipeline. The vertices will be
+# drawn using the given |topology|. A start index of |value| will be used and
+# the count of |count_value| items will be processed.
 RUN <pipeline_name> DRAW_ARRAY INDEXED AS <topology> \
-  START_IDX <value> COUNT <value>
+  START_IDX <value> COUNT <count_value>
 ```
 
 ### Commands
 ```
+# Sets the clear colour to use for |pipeline| which must be a `graphics`
+# pipeline. The colours are integers from 0 - 255.
```

Comment:
what about alpha?

Also, the type of value is intimately tied to the format of the framebuffer.  Maybe add a TODO to revisit this?